assertsprocessor/span_matcher.go

```diff
@@ -69,6 +71,28 @@ func (sm *spanMatcher) getRequest(span *ptrace.Span, serviceKey string) string {
 	return span.Name()
 }
 
+// configListener interface implementation
+func (sm *spanMatcher) isUpdated(prevConfig *Config, currentConfig *Config) bool {
+	updated := !reflect.DeepEqual(prevConfig.RequestContextExps, currentConfig.RequestContextExps)
```

Comment:
FYI: https://github.com/google/go-cmp. In this case, `reflect.DeepEqual` should be fine.